Test/CoreSDK.Test/Net40/TaskExTests.cs

```diff
@@ -0,0 +1,173 @@
+namespace Microsoft.ApplicationInsights
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Assert = Xunit.Assert;
+
+    [TestClass]
+    public class TaskExTests
+    {
+
+        [TestMethod]
+        public void RethrowIfFailedDoesntThrowIfNoExceptionOccured()
+        {
+            var task = Task.Factory.StartNew(() => { });
```

Comment:
Usually "var" is OK as long as the type is clear from the *same* line without specific knowledge of the code (and in complex Linq expressions).
var x = new MyType(); // this is OK
var y = MyFunc(); // Not OK since it requires knowing what MyFunc returns.
this particular one is on the border since Task.Factory.StartNew is "almost" like new task :)